Code/GraphMol/ChemReactions/DaylightParser.cpp

```diff
@@ -87,11 +87,39 @@ std::vector<std::string> splitSmartsIntoComponents(
   }
   return res;
 }
+
+ROMol* constructMolFromString(const std::string &txt,
+    std::map<std::string,std::string> *replacements, bool useSmiles, bool sanitize) {
+  ROMol *mol;
+  if(!useSmiles){
+    mol=SmartsToMol(txt,0,false,replacements);
+    if(sanitize){
+      try{
+        mol->updatePropertyCache();
+        RingInfo *ringInfo=mol->getRingInfo();
+        if(!ringInfo->isInitialized()){
+          ringInfo->initialize();
+        }
+      }
+      catch (const MolSanitizeException &){
+        std::cerr << "Molecule could not be sanitized" << std::endl;
+      }
+    }
+  }
+  else{
+    mol=SmilesToMol(txt,0,sanitize,replacements);
+  }
+  return mol;
+}
+
+
+
+
 }  // end of namespace DaylightParserUtils
 
 ChemicalReaction *RxnSmartsToChemicalReaction(
     const std::string &text, std::map<std::string, std::string> *replacements,
-    bool useSmiles) {
+    bool useSmiles, bool sanitize) {
```

Comment:
I think that is what I mean.   I have found that even .rxn files need to be sanitized before the reactant templates work correctly.  At least the aromaticity does.  I don't think it will hurt anything.

----
Brian Kelley

> On Mar 31, 2016, at 8:29 AM, NadineSchneider <notifications@github.com> wrote:
> 
> In Code/GraphMol/ChemReactions/DaylightParser.cpp:
> 
> >  }  // end of namespace DaylightParserUtils
> >  
> >  ChemicalReaction *RxnSmartsToChemicalReaction(
> >      const std::string &text, std::map<std::string, std::string> *replacements,
> > -    bool useSmiles) {
> > +    bool useSmiles, bool sanitize) {
> So you mean sanitizing the molecules of a reaction always when useSmiles is set to True and otherwise ignoring sanitization? I added this function to improve the drawing of reactions, not sanitizing them leads to ugly images.
> 
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
>